### PR TITLE
credentials/alts: Support UDS addresses in ALTS interop test server

### PR DESCRIPTION
**Pull request by cesarghali**
@roseowl123

### CODE REVIEW

File: interop/alts/server/server.go [hunk 1]
```diff
@@ -30,6 +31,10 @@ import (
 	testpb "google.golang.org/grpc/interop/grpc_testing"
 )
 
+const (
+	udsAddrPrefix = "unix:"
+)
+
 var (
 	hsAddr     = flag.String("alts_handshaker_service_address", "", "ALTS handshaker gRPC service address")
 	serverAddr = flag.String("server_address", ":8080", "The port on which the server is listening")
```
Comment (menghanl) [inline interop/alts/server/server.go]: The usage string is not accurate. It's not a port.

File: interop/alts/server/server.go [hunk 2]
```diff
@@ -38,7 +43,12 @@ var (
 func main() {
 	flag.Parse()
 
-	lis, err := net.Listen("tcp", *serverAddr)
+	// If the server address starts with `unix:`, then we have a UDS address.
+	network := "tcp"
+	if strings.HasPrefix(*serverAddr, udsAddrPrefix) {
+		network = "unix"
+	}
+	lis, err := net.Listen(network, *serverAddr)
```
Comment (menghanl) [inline interop/alts/server/server.go]: I'm not sure, but do you need to also strip "unix:" from the address?
Does `net.Listen("unix", "unix:/tmp/blah")` work?

Comment (cesarghali) [reply interop/alts/server/server.go]: @roseowl123

File: interop/alts/server/server.go (hunk 1 above)
Comment (cesarghali) [reply interop/alts/server/server.go]: Fixed the description.

File: interop/alts/server/server.go (hunk 2 above)
Comment (roseowl123) [reply interop/alts/server/server.go]: net.Listen("unix", "unix:/tmp/blah") would work.

Comment (roseowl123) [reply interop/alts/server/server.go]: Sorry, spoke too soon... I tried net.Listen("unix", "unix:/tmp/blah") in my test and it does not work.

File: interop/alts/server/server.go [hunk 3]
```diff
@@ -30,15 +31,24 @@ import (
...
 )
 
+const (
+	udsAddrPrefix = "unix:"
+)
+
 var (
 	hsAddr     = flag.String("alts_handshaker_service_address", "", "ALTS handshaker gRPC service address")
-	serverAddr = flag.String("server_address", ":8080", "The port on which the server is listening")
+	serverAddr = flag.String("server_address", ":8080", "The address on which the server is listening")
```
Comment (roseowl123) [inline interop/alts/server/server.go]: It's better to explain a little bit about which type and format of address can be accepted here.

File: interop/alts/server/server.go (hunk 2 above)
Comment (cesarghali) [reply interop/alts/server/server.go]: Thanks for checking this. Code is fixed.

File: interop/alts/server/server.go (hunk 3 above)
Comment (cesarghali) [reply interop/alts/server/server.go]: Done.

**Pull request merged**